pipelines/update_exposure_tercile.py

```diff
@@ -0,0 +1,120 @@
+import sys
+from datetime import datetime, timedelta
+
+import numpy as np
+import pandas as pd
+from sqlalchemy import text
+
+from src.utils import database
+
+ROLL_WINDOW = 7
+
+ROLLING_QUERY = text(
+    f"""
+    WITH target_dates AS (
+        SELECT
+            pcode,
+            adm_level,
+            valid_date,
+            sum as original_sum
+        FROM app.floodscan_exposure
+        WHERE EXTRACT(MONTH FROM valid_date) = :month
+            AND EXTRACT(DAY FROM valid_date) = :day
+    ),
+    date_ranges AS (
+        SELECT
+            t.pcode,
+            t.adm_level,
+            t.valid_date as target_date,
+            t.original_sum as sum,
+            d.valid_date,
+            d.sum as daily_sum
+        FROM target_dates t
+        JOIN app.floodscan_exposure d
+            ON d.pcode = t.pcode
+            AND d.valid_date BETWEEN t.valid_date - INTERVAL '{ROLL_WINDOW-1} days' AND t.valid_date
+    )
+    SELECT
+        pcode,
+        adm_level,
+        target_date as valid_date,
+        AVG(daily_sum) as rolling_avg
+    FROM date_ranges
+    GROUP BY pcode, adm_level, target_date
+    ORDER BY pcode, target_date;
+    """
+)
+
+
+def assign_tercile(row, boundaries):
+    """
+    Assign tercile values based on boundaries.
+    Returns:
+        -1: Below lower tercile
+        0: Between terciles
+        1: Above upper tercile
+    """
+    pcode_bounds = boundaries.loc[row["pcode"]]
+    value = row["rolling_avg"]
+    if value < pcode_bounds["lower_tercile"]:
```

Comment:
This is a bit of shortcut that I'll need to fix but want to get your take. 

If the lower and upper tercile are the same (likely 0), I want to classify an observation that matches both to be in the "average" band. This accomplishes that but now we have a bug if  `value = pcode_bounds["lower_tercile"]` and `value != pcode_bounds["upper_tercile"]` --> it'll get dumped into the final `else` statement. 

Would you handle the 0 case any differently? Otherwise I'll adjust the logic a bit here to try and avoid that bug.

Reply:
I think this is the best way to handle it yeah!